fog/ingest/server/src/controller.rs

```diff
@@ -469,7 +469,19 @@ where
                             }
                             break;
                         }
+                        Err(Error::PublishReport) => {
+                            // This is already logged adequately within
+                            // self.publish_report
+                            //
+                            // Failing to publish a report is not fatal, because
+                            // we attempt to publish
+                            // on every block, and even if it only succeeds half
+                            // the time, it wont make much difference in the
+                            // pubkey expiry window.
+                        }
```

Comment:
I feel like this would be the simplest  way to handle this
```suggestion
                        // catch publish errors to prevent log duplication
                        Err(Error::PublishReport) => _ ,
```

Reply:
I think this has a problem which is that, we will no longer sleep when `PublishReport` errors occur. that seems not great, it's usually better to have some kind of backoff if there is a connection error (which is what this is usually)

is this actually what we did in the other repo?

Reply:
I didn't notice the sleep right there - good point. Given the OK case breaks out of the loop above, the sleep could just be moved outside of the match statement and added to the end.

Reply:
So something like: 

```rust
                loop {
                    match self.publish_report(&ingress_pubkey, &mut state) {
                        Ok(ingress_key_status) => {
                            // If our key is retired, and the index we want to scan is past expiry,
                            // early return. Note, we don't even NEED to scan
                            // when block.index == pubkey_expiry, because the
                            // semantic of pubkey expiry is that it bounds the tombstone block, and
                            // a transaction cannot land in its
                            // tombstone block. But scanning the
                            // tombstone block as well may help
                            // deal with off-by-one errors somewhere else, and doesn't really hurt.
                            if ingress_key_status.retired
                                && block.index > ingress_key_status.pubkey_expiry
                            {
                                log::warn!(self.logger, "When preparing to process block index {}, we discovered that our ingress key is expired: {:?}. Switching to idle and nuking keys.", block.index, ingress_key_status);
                                state.set_idle();
                                self.new_egress_key(&mut state)
                                    .expect("Failure to rotate egress key can't be recovered from");
                                return;
                            }
                            break;
                        }
                        Err(Error::PublishReport) => {}
                        Err(err) => {
                            log::error!(
                                self.logger,
                                "Could not publish ingest report at block index {}, will retry: {}",
                                block.index,
                                err
                            );
                        }
                    }
                    std::thread::sleep(std::time::Duration::from_secs(retry_seconds));
                    retry_seconds = std::cmp::min(retry_seconds + 1, 30);
```

Reply:
Or we just wrap the `error!` call in an `if err != Error::PublishReport`

Reply:
I don't think we should reduce the context provided by the comment - without it it's very unclear why we should treat PublishReport separately.

I have restored the sleep in the PublishReport case.